app/src/main/java/com/rootstrap/android/ui/activity/main/SignInActivityViewModel.kt

```diff
@@ -13,30 +13,40 @@ import com.rootstrap.android.util.ViewModelListener
 import com.rootstrap.android.util.extensions.ApiErrorType
 import com.rootstrap.android.util.extensions.ApiException
 import kotlinx.coroutines.launch
+import java.io.IOException
 
-open class SignInActivityViewModel(listener: ViewModelListener?) : BaseViewModel(listener) {
-
-    private val manager: IUserManager = UserManager
+open class SignInActivityViewModel(
+    listener: ViewModelListener?,
+    private var manager: IUserManager
+) : BaseViewModel(listener) {
 
     var state: SignInState = SignInState.none
         set(value) {
             field = value
             listener?.updateState()
         }
 
+    fun canSignIn(userName: String?, password: String?): Boolean {
+        return userName.isNullOrEmpty().not() && password.isNullOrEmpty().not()
```

Comment:
add password validation: password length should be more than 6 characters.

Reply:
done!